src/miner.cpp

```diff
@@ -387,23 +357,14 @@ CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const MinerAddre
             if (nBlockSigOps + nTxSigOps >= MAX_BLOCK_SIGOPS)
                 continue;
 
-            // Skip free transactions if we're past the minimum block size:
             const uint256& hash = tx.GetHash();
-            double dPriorityDelta = 0;
-            CAmount nFeeDelta = 0;
-            mempool.ApplyDeltas(hash, dPriorityDelta, nFeeDelta);
-            if (fSortedByFee && (dPriorityDelta <= 0) && (nFeeDelta <= 0) && (feeRate < ::minRelayTxFee) && (nBlockSize + nTxSize >= nBlockMinSize))
+            CAmount nFeeDelta = mempool.GetFeeDelta(hash);
+
+            // Skip free transactions if we're past the priority block size:
+            if ((nFeeDelta <= 0) && (feeRate < ::minRelayTxFee) && (nBlockSize + nTxSize >= nBlockPrioritySize))
```

Comment:
This does not maintain the current fee behaviour. Revert to this behaviour:
```suggestion
            // Skip free transactions if we're past the minimum block size:
            CAmount nFeeDelta = mempool.GetFeeDelta(hash);
            if ((nFeeDelta <= 0) && (feeRate < ::minRelayTxFee) && (nBlockSize + nTxSize >= nBlockMinSize))
```

Reply:
by default, `nBlockMinSize` is set to `DEFAULT_BLOCK_MIN_SIZE = 0` (see `policy.h`.) this means that by default, we always skip free transactions (if there is no `nFeeDelta`). this causes many of our RPC tests to fail: for example, in `wallet_changeaddresses.py`, we call `z_shieldcoinbase(from, to, 0)` with `fee = 0`. 

i propose retaining `nBlockMinSize` here, but changing `DEFAULT_BLOCK_MIN_SIZE = DEFAULT_BLOCK_MAX_SIZE/2` in `policy.h`. the alternative would be to manually set `-blockminsize` in every test.

Reply:
in `policy.h` there's also the following comment about `DEFAULT_BLOCK_PRIORITY_SIZE = DEFAULT_BLOCK_MAX_SIZE/2`:

```c++
/** Default for -blockprioritysize, maximum space for zero/low-fee transactions **/
```

since `-blockprioritysize` was meant to set the maximum space for zero fee transactions, i originally thought it was apt to use it here.